### PR TITLE
feat: Add DS Guides link to component readmes

### LINKED ISSUE #8149: Add DS Guides to each readme in component-library
- Add the [DS Guide link](https://www.notion.so/MetaMask-Design-System-Guides-Design-f86ecc914d6b4eb6873a122b83c12940) to each component's readme

### PR DESCRIPTION
**Pull request by brianacnguyen**
## **Description**
- Add DS Guides link to component readmes

## **Related issues**

Fixes: #8149 

## **Manual testing steps**

1. Go to a component's readme 
2. Click on title
3.

### **After**
Screenshot 2024-01-25 at 8 51 23 AM
Screenshot 2024-01-25 at 8 51 44 AM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by georgewrmarshall — COMMENTED** [review]
Links i've checked so far are working well and this approach would certainly benefit our internal contributors. However, to enhance it further, we could consider updating the links to point to the public-facing versions of these Notion pages. I've previously received feedback about including private Figma links. Now, our Figma libraries for components and color are openly viewable by the public. As an open-source project, it's crucial for MetaMask to foster inclusivity and transparency in our documentation and processes. I'm also wondering if we could be a little more intentioned about where the link appears. Instead of hyperlinking the title could we add the link underneath the header e.g.

# Accordion
 [Design documentation](https://southern-bush-399.notion.site/Accordion-f43e500b92754d9bb06bc72553c44494)
 
 
Screenshot 2024-01-25 at 1 15 55 PM


I'm also comfortable to approve this and create a ticket to improve these pages

**Comment by brianacnguyen** [comment]
> Links i've checked so far are working well and this approach would certainly benefit our internal contributors. However, to enhance it further, we could consider updating the links to point to the public-facing versions of these Notion pages. I've previously received feedback about including private Figma links. Now, our Figma libraries for components and color are openly viewable by the public. As an open-source project, it's crucial for MetaMask to foster inclusivity and transparency in our documentation and processes. I'm also wondering if we could be a little more intentioned about where the link appears. Instead of hyperlinking the title could we add the link underneath the header e.g.
> 
> # Accordion
> [Design documentation](https://southern-bush-399.notion.site/Accordion-f43e500b92754d9bb06bc72553c44494)
> 
> Screenshot 2024-01-25 at 1 15 55 PM
> 
> I'm also comfortable to approve this and create a ticket to improve these pages

I actually think having the link as part of the title works better because it reduces the noise/space the viewer will have to scroll to see. The more we add on top, the more the content gets pushed down, and less people will care about the readme

**Review by georgewrmarshall — DISMISSED** [review]

File: app/component-library/components/BottomSheets/BottomSheetFooter/README.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
-# BottomSheetFooter
-
+# [BottomSheetFooter](https://metamask-consensys.notion.site/Footer-b0eacc3951bf4962b75f6c4f95a8c2df)
```
Comment (georgewrmarshall) [inline app/component-library/components/BottomSheets/BottomSheetFooter/README.md:1]: This goes to the Footer page I suppose there is no `BottomSheetFooter` equivalent in design?
Screenshot 2024-02-05 at 4 31 52 PM

**Review by georgewrmarshall — APPROVED** [review]
LGTM 🚀 
- Checked all links go to correct notion page

File: app/component-library/components/BottomSheets/BottomSheetFooter/README.md (hunk 1 above)
Comment (brianacnguyen) [reply app/component-library/components/BottomSheets/BottomSheetFooter/README.md:1]: not that I have found

**Pull request merged**